component/attr_cache/attr_cache.go

```diff
@@ -566,15 +512,25 @@ func (ac *AttrCache) IsDirEmpty(options internal.IsDirEmptyOptions) bool {
 }
 
 func (ac *AttrCache) anyContentsInCache(prefix string) bool {
-	// Add a trailing / so that we only find child paths under the directory and not paths that have the same prefix
-	prefix = dirToPrefix(prefix)
 	ac.cacheLock.RLock()
 	defer ac.cacheLock.RUnlock()
-	for key, value := range ac.cacheMap {
-		if strings.HasPrefix(key, prefix) && value.valid() && value.exists() {
+
+	cachedContentItem, err := ac.cacheMap.get(prefix)
+	if err != nil {
+		return false
+	} else { //TODO: this isn't looking at each child or not going through the sub tree. is that an issue?
+		if cachedContentItem.exists() {
+			for _, chldCachedContentItem := range cachedContentItem.children {
+				if chldCachedContentItem.exists() {
+					ac.anyContentsInCache(chldCachedContentItem.attr.Path)
```

Comment:
```suggestion
					return true
```
This should not be recursive.
As soon as one of the children is found to exist, return true.